### ISSUE TITLE
Is it possible to Memoize styled components to prevent unnecessary re-render?

### ISSUE THREAD

**Issue by roldanjrCodeArts9711** [callstack/linaria#538]
## Is it possible to Memoize styled components to prevent unnecessary re-render to optimize the performance of the app?

Here is the screenshot of React Profiler that shows components that repeatedly re-rendered.
Thank you for any help.

![image](https://user-images.githubusercontent.com/38913069/74088336-446f0180-4ad0-11ea-821d-e5acb50bfbcd.png)

**Comment by satya164** [comment]
Usually you use DOM elements with `styled` such as `styled.div` and pass `children` prop to it. The `children` prop changes each render, so if they use `React.memo`, it essentially doesn't do anything except adding extra overhead when checking props.

Also wrapping every component in `React.memo` can add a lot of overhead by having to memoize a lot of things and extra comparision for a lot of things that actually do change. You should instead memoize the components where you use these styled components instead, which will have a lot less overhead and can actually work.

**Comment by roldanjrCodeArts9711** [comment]
Thanks @satya164. I managed it by separating and grouping those `StyledComponents` that doesn't have props that constantly change every time and wrapped it with `React.memo` so it won't get re-rendered if not needed.

Also, I separated those `StyledComponents` that have state inside to ensure that it will not involve other components to be re-rendered too if its state change.

Here is the new screenshot,

![image](https://user-images.githubusercontent.com/38913069/74136537-2c5ec580-4c29-11ea-80c6-2954efac6ff2.png)
